server/src/main/java/org/elasticsearch/action/ActionListenerImplementations.java

```diff
@@ -24,6 +26,8 @@
  */
 class ActionListenerImplementations {
 
+    private static final Cleaner cleaner = Cleaner.create();
```

Comment:
The recommendation seemed to be that you have a single `Cleaner` per library. I wonder if we should implement `ElasticSearchCleaner.get()` so that `LeakTracker` and this can share one?

Reply:
It would be really nice to integrate it with a test life-cycle.

Reply:
I did have a chat with @ywangd about this and he suggested a goal might be to run a node with assertions turned on, so presumably it would nice for it to work in a test lifecycle (trigger assertions at the end of the test that all these handlers created during that test have been completed) but also to kill a node running with assertions turned on when one of these handlers gets garbage collected without being completed (what I think it does now)